apollo-router-core/src/services/http_compat.rs

```diff
@@ -30,14 +94,18 @@ impl<T> Request<T> {
     ///
     /// Required parameters are required in non-testing code to create a Request.
     pub fn new(
-        headers: Vec<(HeaderName, HeaderValue)>,
+        headers: MultiMap<IntoHeaderName, IntoHeaderValue>,
```

Comment:
why use `MultiMap` and not [`HeaderMap`](https://docs.rs/http/latest/http/header/struct.HeaderMap.html)?

Reply:
If we use `HeaderMap` then we can't use `IntoHeaderName` and `IntoHeaderValue`. In this implementation it's required us to late convert in the call to `build`.

Hyper also has so do some strange stuff to get around the fallibleness of header construction for late building.
https://github.com/hyperium/http/blob/master/src/request.rs#L1033-L1053

Reply:
alright then